uxarray/core/zonal.py

```diff
@@ -18,23 +17,15 @@ def _compute_non_conservative_zonal_mean(uxda, latitudes, use_robust_weights=Fal
         # Create a NumPy array for storing results
         result = np.zeros(shape, dtype=uxda.dtype)
 
-    faces_edge_nodes_xyz = _get_cartesian_face_edge_nodes(
-        uxgrid.face_node_connectivity.values,
-        uxgrid.n_face,
-        uxgrid.n_max_face_nodes,
-        uxgrid.node_x.values,
-        uxgrid.node_y.values,
-        uxgrid.node_z.values,
-    )
-
     bounds = uxgrid.bounds.values
+    faces_edges_cartesian = uxgrid.faces_edges_cartesian.values
 
     for i, lat in enumerate(latitudes):
         face_indices = uxda.uxgrid.get_faces_at_constant_latitude(lat)
 
         z = np.sin(np.deg2rad(lat))
 
-        faces_edge_nodes_xyz_candidate = faces_edge_nodes_xyz[face_indices, :, :, :]
+        faces_edge_nodes_xyz_candidate = faces_edges_cartesian[face_indices, :, :, :]
```

Comment:
Wouldn't a better implementation here be to only construct the `faces_edges_cartesian` for the indices that we require, instead of needing to construct the entire array and then index?

For each line of constant latitude, typically only a small percentage of the total faces will be candidates.

Reply:
> Something like the following should work if we put it inside of the loop?
> 
> ```python
> _get_cartesian_face_edge_nodes(
>        uxgrid.face_node_connectivity.values[face_indices], # only candidate faces
>        len(face_indices), # number of candidate faces
>        uxgrid.n_max_face_nodes,
>        uxgrid.node_x.values,
>        uxgrid.node_y.values,
>        uxgrid.node_z.values,
>     )
> ```

Please refer to my comment below: the lalon bounds building will required this connectivity for the entire grid anyway